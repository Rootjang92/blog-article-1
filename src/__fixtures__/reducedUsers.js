const reducedUsers = [
  {
    id: 1,
    first_name: "Diana",
    last_name: "Prince",
    email: "dianaprince@flatley.com",
    nickname: "Wonder Woman"
  },
  {
    id: 2,
    first_name: "Bruce",
    last_name: "Wayne",
    email: "brucewayne@cummerata.com",
    nickname: "Batman"
  }
];

export default reducedUsers;