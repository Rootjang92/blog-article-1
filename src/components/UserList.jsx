import React from "react";

const UserList = ({ users }) => (
  <table>
    <thead>
      <tr>
        <td>Full Name</td>
        <td>Email</td>
        <td>Nickname</td>
      </tr>
    </thead>
    <tbody>
      {users.map(user => {
        const { id, first_name, last_name, email, nickname } = user;
        return (
          <tr className="User" key={id}>
            <td>
              {first_name} {last_name}
            </td>
            <td>{email}</td>
            <td>{nickname}</td>
          </tr>
        );
      })}
    </tbody>
  </table>
);

export default UserList;
