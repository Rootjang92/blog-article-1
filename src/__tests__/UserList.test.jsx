import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import UserList from "components/UserList";
import reducedUsers from "__fixtures__/reducedUsers";

describe("UserList", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<UserList users={reducedUsers} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders a list of rows with users", () => {
    const componentWrapper = shallow(<UserList users={reducedUsers} />);
    const numberOfUserRows = componentWrapper.find("tr.User").length;

    expect(numberOfUserRows).toEqual(2);
  });
});
