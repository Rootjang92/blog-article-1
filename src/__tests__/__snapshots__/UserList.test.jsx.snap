// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <td>
        Full Name
      </td>
      <td>
        Email
      </td>
      <td>
        Nickname
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      className="User"
    >
      <td>
        Diana
         
        Prince
      </td>
      <td>
        dianaprince@flatley.com
      </td>
      <td>
        Wonder Woman
      </td>
    </tr>
    <tr
      className="User"
    >
      <td>
        Bruce
         
        Wayne
      </td>
      <td>
        brucewayne@cummerata.com
      </td>
      <td>
        Batman
      </td>
    </tr>
  </tbody>
</table>
`;
